core/src/impl/Kokkos_Core.cpp

```diff
@@ -99,6 +99,7 @@ setenv("MEMKIND_HBW_NODES", "1", 0);
   if (use_gpu < 0 && ndevices >= 0) {
     auto local_rank_str = std::getenv("OMPI_COMM_WORLD_LOCAL_RANK"); //OpenMPI
```

Comment:
Our Slurm environment does not set it when using, for example, `srun` to start the application and there are other places where this is also mentioned, e.g., https://narkive.com/CT2zjQKV.6 or https://github.com/baidu-research/baidu-allreduce/blob/73c7b7f8d34f7163a3a46aa0fe48e2a418d25fe9/test/test.cpp#L71
I can confirm that this change works on our cluster, though it might be worth adding some more logic in general to this option (if people think it's worth it).